src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -42,13 +42,26 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
     /// </summary>
     /// <typeparam name="T">The resource type.</typeparam>
     /// <param name="builder">The resource builder.</param>
-    /// <param name="source">The source name of the volume.</param>
-    /// <param name="target">The target path where the file or directory is mounted in the container.</param>
-    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <param name="name">The name of the volume.</param>
+    /// <param name="target">The target path where the volume is mounted in the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if the volume should be mounted as read-only.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithVolume<T>(this IResourceBuilder<T> builder, string name, string target, bool isReadOnly = false) where T : ContainerResource
+    {
+        var annotation = new ContainerMountAnnotation(name, target, ContainerMountType.Named, isReadOnly);
```

Comment:
If we make this change, then `ContainerMountType.Named` should be renamed.

Options:
* Rename to `ContainerMountType.Volume`
* Add `ContainerMountType.Anonymous` and use it from this method.

I believe `Named` is passed to DCP. Either DCP should change values it accepts to something that makes more sense, or we leave DCP and just map from our value to the DCP value, i.e. Volume -> Named.

Reply:
DCP models its support for mounts aligned with the lower level `--mount` flag  in Docker, which covers all mount types, while the higher level concepts like [bind mounts](https://docs.docker.com/storage/bind-mounts/) and [volumes](https://docs.docker.com/storage/volumes/) are what we're aligned with for these APIs. 

I think the best approach is your first suggestion, i.e. we update `ContainerMountType.Named` to `ContainerMountType.Volume` and update our mapping to DCP.